src/RecyclableMemoryStream.cs

```diff
@@ -610,7 +610,7 @@ async Task CopyToAsyncImpl()
                 }
                 else
                 {
-                    return destination.WriteAsync(this.largeBuffer, 0, this.length, cancellationToken);
+                    return destination.WriteAsync(this.largeBuffer, 0, (int)this.length, cancellationToken);
```

Comment:
We should add `Debug.Assert` assertions everywhere `length` is assumed to be in the `int` boundary, just in case.